spec/lib/vanagon/driver_spec.rb

```diff
@@ -54,6 +54,40 @@ def create_driver(platform, options = {})
                               'engine' => 'hardware' })
     end
 
+    it 'returns the platform and a scheduler engine when scheduled resources are enabled and an engine is specified' do
+      ENV['VANAGON_SCHEDULED_HARDWARE'] = "true"
+
+      platform = eval_platform('aix-7.1-ppc', <<-END)
+        platform 'aix-7.1-ppc' do |plat|
+          plat.build_host ["pe-aix-71-01", "pe-aix-71-02"]
+          plat.engine "scheduled"
+        end
+      END
+
+      info = create_driver(platform).build_host_info
+
+      expect(info).to match({ 'name'   => 'aix-7.1-ppc',
```

Comment:
Lemme see if I've got my head around everything:

We're probably all doomed to be sad about platform tagging and naming.  😿 

We did agree to use vmpooler names for interoperating with the ci.next scheduler (always-be-scheduling).  That means that we can either translate platform names in vanagon when they are outbound to always-be-scheduling or we can translate on the inbound side (note that the scheduler is sometimes a step or few removed from the output of `build_host_info` when we are running in a pipeline).

The work here started as "let's get `build_host_info` to not allocate and lock/unlock hardware" which implies "let's get vanagon on a path to not allocate / lock / unlock hardware". 

The logic would be to trigger `build_host_info` in such a way that we know ABS is asking for what platform a build is going to use, and if it's hardware to provide a hardware platform name and a count (rather than allocating and providing a hostname, as is the current flow).  The scheduler does the needful, and comes back with a specific hostname and provides that to vanagon when it's time to actually do the build.

We've so far never output a platform or template name for hardware from `build_host_info`, which means that whatever we do here, we're going to be changing behavior (under the scope of a specific engine being created) and will provide some version of a platform name.  I think @joshcooper's point (that I sympathize with) is that, if we're suddenly providing a platform name, and we know the recipient of that platform name wants it as a vmpooler-style platform name, we should probably actually output it in that format.

We are considering going a step further here and expanding the 'scheduler' engine (probably will be 'always-be-scheduling' to make things clearer) to also return template names for vmpooler builds, which, presumably would also be output in vmpooler format (as that's what always-be-scheduling would prefer to consume).  I think that's mostly a passthrough.